packages/fuels-programs/src/contract.rs

```diff
@@ -86,7 +86,7 @@ impl Default for CallParameters {
     fn default() -> Self {
         Self {
             amount: DEFAULT_CALL_PARAMS_AMOUNT,
-            asset_id: BASE_ASSET_ID,
+            asset_id: Default::default(),
```

Comment:
Same concern as for ResourceFilter

Reply:
This is now updated to use `Option<AssetId>`. If the `AssetId` is not set the provider will use `base_asset_id`